src/Conversion/ONNXToTorch/NN/LeakyReluOp.cpp

```diff
@@ -55,46 +51,31 @@ using namespace mlir::torch::Torch;
  *
  */
 
-class ONNXLeakyReluOpToTorchLowering : public ConversionPattern {
+class ONNXLeakyReluOpToTorchLowering : public OpConversionPattern<ONNXLeakyReluOp> {
 public:
-  ONNXLeakyReluOpToTorchLowering(TypeConverter &typeConverter, MLIRContext *ctx)
-      : ConversionPattern(
-            typeConverter, mlir::ONNXLeakyReluOp::getOperationName(), 1, ctx) {}
-
-  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
-      ConversionPatternRewriter &rewriter) const final {
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXLeakyReluOp op, OpAdaptor adaptor,
+        ConversionPatternRewriter &rewriter) const override {
 
-    Location loc = op->getLoc();
+    Location loc = op.getLoc();
     mlir::MLIRContext *context = op->getContext();
-    ONNXLeakyReluOp op1 = llvm::dyn_cast<ONNXLeakyReluOp>(op);
-    ONNXLeakyReluOpAdaptor adapter(op1);
-
-    Value x = op1.X();
 
-    auto alpha = adapter.alphaAttr(); // mlir::FloatAttr
+    auto x = adaptor.X();
+    auto alpha = adaptor.alphaAttr(); // mlir::FloatAttr
     auto negSlope = alpha.getValue(); // APSFloat
     auto negSlopeFloatValue = FloatAttr::get(
-        mlir::FloatType::getF64(op->getContext()), negSlope.convertToFloat());
+        mlir::FloatType::getF64(op.getContext()), negSlope.convertToFloat());
     Value negSlopeConstFloat =
```

Comment:
This whole conversion chain leading up to here is really sketchy. There should never be a reason to construct an APFloat from a double obtained from an APFloat. If that kind of NEAREST_EVEN behavior is desired, an explicit APFloat conversion should be used.

Reply:
`negSlope` is not a double. These are `f32` types so we can probably keep this. I saw similar constructs in `onnx-mlir` itself. It would be nice to have something cleaner though.

https://github.com/onnx/onnx-mlir/blob/15f75fc840f2c2664da87500693c162e0de17d06/src/Dialect/ONNX/ONNXOpsHelper.cpp#L385